x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/plan/physical/AggregateExec.java

```diff
@@ -184,6 +187,25 @@ protected AttributeSet computeReferences() {
         return mode.isInputPartial() ? new AttributeSet(intermediateAttributes) : Aggregate.computeReferences(aggregates, groupings);
     }
 
+    /** Returns the attributes that can be loaded from ordinals -- no explicit extraction is needed */
+    public List<Attribute> ordinalAttributes() {
+        List<Attribute> orginalAttributs = new ArrayList<>(groupings.size());
+        if (groupings().size() == 1) {
+            // CATEGORIZE requires the standard hash aggregator as well.
+            if (groupings.get(0).anyMatch(e -> e instanceof Categorize) == false) {
+                var leaves = new LinkedList<>();
+                // TODO: this seems out of place
+                aggregates.stream().filter(a -> groupings.contains(a) == false).forEach(a -> leaves.addAll(a.collectLeaves()));
```

Comment:
Lets use collect instead of `addAll` in `forEach`. 
Since it is used for `.contains` check below, should it collect to set instead of the list?